go-controller/pkg/clustermanager/default_network_cluster_controller.go

```diff
@@ -0,0 +1,400 @@
+	}
+
+	return ncm.defaultNetworkController.deleteNode(node)
+}
+
+func (ncm *defaultNetworkClusterController) syncNodes(nodes []interface{}) error {
+	for _, tmp := range nodes {
+		node, ok := tmp.(*corev1.Node)
+		if !ok {
+			return fmt.Errorf("spurious object in syncNodes: %v", tmp)
+		}
+
+		hostSubnets, _ := util.ParseNodeHostSubnetAnnotation(node, ovntypes.DefaultNetworkName)
+		if config.HybridOverlay.Enabled && len(hostSubnets) == 0 && houtil.IsHybridOverlayNode(node) {
+			// this is a hybrid overlay node so mark as allocated from the hybrid overlay subnet allocator
+			hostSubnet, err := houtil.ParseHybridOverlayHostSubnet(node)
+			if err != nil {
+				klog.Errorf("Failed to parse hybrid overlay for node %s: %w", node.Name, err)
+			} else if hostSubnet != nil {
+				klog.V(5).Infof("Node %s contains subnets: %v", node.Name, hostSubnet)
+				if err := ncm.hybridOverlaySubnetAllocator.MarkSubnetsAllocated(node.Name, hostSubnet); err != nil {
+					klog.Errorf("Failed to mark the subnet %v as allocated in the hybrid subnet allocator for node %s: %w", hostSubnet, node.Name, err)
+				}
+			}
+		}
+	}
+
+	return ncm.defaultNetworkController.syncNodes(nodes)
+}
+
+func (h *defaultNetworkClusterControllerEventHandler) AreResourcesEqual(obj1, obj2 interface{}) (bool, error) {
+	// switch based on type
+	switch h.objType {
+	case factory.NodeType:
+		node1, ok := obj1.(*corev1.Node)
+		if !ok {
+			return false, fmt.Errorf("could not cast obj1 of type %T to *corev1.Node", obj1)
+		}
+		node2, ok := obj2.(*corev1.Node)
+		if !ok {
+			return false, fmt.Errorf("could not cast obj2 of type %T to *corev1.Node", obj2)
+		}
+
+		// when shouldUpdateNode is false, the hostsubnet is not assigned by ovn-kubernetes
+		shouldUpdate, err := util.ShouldUpdateNode(node2, node1)
+		if err != nil {
+			klog.Errorf(err.Error())
+		}
+		return !shouldUpdate, nil
+	}
+
+	return false, nil
+}
+
+// GetInternalCacheEntry returns the internal cache entry for this object
+func (h *defaultNetworkClusterControllerEventHandler) GetInternalCacheEntry(obj interface{}) interface{} {
+	return nil
+}
+
+// getResourceFromInformerCache returns the latest state of the object from the informers cache
+// given an object key and its type,
+func (h *defaultNetworkClusterControllerEventHandler) GetResourceFromInformerCache(key string) (interface{}, error) {
+	var obj interface{}
+	var name string
+	var err error
+
+	_, name, err = cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil, fmt.Errorf("failed to split key %s: %v", key, err)
+	}
+
+	switch h.objType {
+	case factory.NodeType,
+		factory.EgressNodeType:
+		obj, err = h.ncm.watchFactory.GetNode(name)
+
+	default:
+		err = fmt.Errorf("object type %s not supported, cannot retrieve it from informers cache",
+			h.objType)
+	}
+	return obj, err
+}
+
+// RecordAddEvent records the add event on this object. Not used here.
+func (h *defaultNetworkClusterControllerEventHandler) RecordAddEvent(obj interface{}) {
```

Comment:
can you check my comment here?:
https://github.com/ovn-org/ovn-kubernetes/pull/3127#discussion_r1089986827
move these and 
```
// GetInternalCacheEntry returns the internal cache entry for this object
func (h *defaultNetworkClusterControllerEventHandler) GetInternalCacheEntry(obj interface{}) interface{} {
	return nil
}
```
to end of file, so that the useful functions are outlined first.